app/controllers/pieces_controller.rb

```diff
@@ -2,13 +2,27 @@ class PiecesController < ApplicationController
   before_action :authenticate_user!
   def update
     @piece = Piece.find(params[:id])
+    origin_row = @piece.row_coordinate
+    origin_col = @piece.column_coordinate
     move_result = check_move_validity(@piece)
     return render json: { update_attempt: 'invalid move' } unless move_result
 
     in_check = @piece.game.in_check?.present?
     stalemate = @piece.game.stalemate?(@piece.color)
     pawn_to_promote = pawn_promotion(@piece)
 
+    game_channel = 'game_channel-' + @piece.game.id.to_s
+
+    Pusher.trigger(game_channel, move_result, message: 'hello world',
+                                              current_user: current_user.id,
+                                              color_moved: @piece.color,
+                                              origin_square: { row: origin_row,
+                                                               col: origin_col },
+                                              destination_square: { row: @piece.row_coordinate,
+                                                                    col: @piece.column_coordinate },
+                                              stalemate: stalemate,
+                                              in_check: in_check)
```

Comment:
A nice way to test this would be to use `allow` and `have_received`.

With 3rd party libraries, we don't need to test that they work in our test suite (they have their own tests). But we should test that we are passing the correct params to them. 

```
allow(Pusher).to receive(:trigger)

expect(Pusher).to have_received(:trigger).with('game_channel-1', ect, ect)
```

Since this is an important part of the app, having a test that the correct arguments are being passed to Pusher will add confidence that we'll never break it.